fuels-signers/src/lib.rs

```diff
@@ -112,9 +111,8 @@ mod tests {
         let signature = wallet.sign_transaction(&mut tx).await.unwrap();
         let message = Message::new(tx.id());
 
-        // TODO(oleksii): impl FromStr for fuel_crypto::Signature
         // Check if signature is what we expect it to be
-        // assert_eq!(signature.compact, Signature::from_str("0xa1287a24af13fc102cb9e60988b558d5575d7870032f64bafcc2deda2c99125fb25eca55a29a169de156cb30700965e2b26278fcc7ad375bc720440ea50ba3cb").unwrap().compact);
+        assert_eq!(signature, Signature::from_str("0x87493a4e6e02e70e6a380f9c9ad2167cc01ccd6cf058c371316911c80e2415c8b7a406268b09b985581ebbe70de81ff59bba99d24dae5295d0a137b7b9a8507c").unwrap());
```

Comment:
Please note that the signature here has changed. Not sure if it's normal. `sign_and_verify` didn't change, while `sign_tx_and_verify` did.

Reply:
But we have to "prove" to ourselves that it is because the tx format changed

Reply:
Well, the sign/recover/verify workflow is intact…

>  the transaction format/content changed

Then the old  `secp256k1`-based tests should have caught it, I think?

Reply:
It's interesting that this is the only place the signature changed, while the signature on the hardcoded message "my message" above wasn't affected. Definitely seems like a change in the tx id would've affected this signature earlier though.

Reply:
So what should we do? IMO the 4 options are:
- Previous implementation was faulty, and this is "the correct one" for some obscure elliptic-curve level reason
- The new implementation is faulty for the the same kind of reason
- The tx_id format changed, and for some reason it was not caught by the previous tests
- There is a dependency/version issue that for instance would prevent the tests to detect the change

Reply:
Maybe let's merge master back into it just to make sure it's not deps/version issue (I'm shooting blindfolded haha)

Reply:
Figured it out. Message::new() from fuel-crypto always hashes the input. Before this PR, we skipped hashing the tx id since it was already guaranteed to be 32 bytes. The reason why the signature in the other test didn't change is that we had a step in fuels-rs [to hash the message](https://github.com/FuelLabs/fuels-rs/blob/master/fuels-signers/src/wallet.rs#L227), which matches the new behavior of Message::new().

If we want to restore the original behavior, we'd have to use the "unsafe" method `Message::from_bytes_unchecked`. This may be appropriate since we already know a transaction id comes from the output of a hash function.

Reply:
@Voxelot Thank you!

Reply:
Alternatively, you can create a message from a [Hasher](https://docs.rs/fuel-crypto/0.4.2/fuel_crypto/struct.Message.html#impl-From%3C%26%27_%20Hasher%3E)